driver/src/main/java/org/neo4j/driver/internal/SimpleRecord.java

```diff
@@ -57,7 +58,16 @@ public Value get( int fieldIndex )
     @Override
     public Value get( String fieldName )
     {
-        return fields[fieldLookup.get( fieldName )];
+        Integer fieldIndex = fieldLookup.get( fieldName );
+
+        if ( fieldIndex == null )
+        {
+            throw new ValueNotFoundException( fieldName );
```

Comment:
How come we have a discrepancy between this get() method and the property() method on the Entity interface? It seems it'd be easier for the user if accessing a non-existent field were the same in both cases.

@boggle argued we should return nulls in these cases, since that is what cypher does. I'm inclined to agree - a null check is nicer than exceptions for control flows, and if the user code is not equipped to deal with it they will get an exception anyway. Eg: I argue we should just return nulls from all of these rather than throw an exception, and update the javadocs and add tests accordingly.